src/Client.php

```diff
@@ -67,6 +70,8 @@ class Client
 
     private array $typeMap;
 
+    private Encoder $builderEncoder;
```

Comment:
PHP 8.1 allows us to make this readonly:
```suggestion
    private readonly Encoder $builderEncoder;
```

Reply:
Also, just thought of this after seeing the Psalm errors in the follow-up PR, but specifying a template for the Encoder might be helpful:
```suggestion
    /** @psalm-var Encoder<array|stdClass|Document|PackedArray, mixed> */
    private readonly Encoder $builderEncoder;
```